From 669f75c8fb16337c9e30c4b3894917110235483e Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Tue, 16 Aug 2016 17:52:34 +0200
Subject: [PATCH 0451/1345] arm64: dts: marvell: add pinctrl description for
 I2C0 in AP806 dtsi

commit  80595273da6aad19752629fad207bed6ade5e0c4 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

The AP806's MPP4 and MPP5 can be used as I2C0 - add appropriate description
and use it for Armada 8K.

Change-Id: Ibb5dbd7322a59bf49b3e5c6559f70dd5fbb98676
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32377
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi |    2 ++
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi   |    5 +++++
 2 files changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 2011292..5393436 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -64,6 +64,8 @@
 			i2c@511000 {
 				status = "okay";
 				clock-frequency = <100000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&ap_i2c0_pins>;
 				eeprom@57 {
 					compatible = "at24,24c64";
 					reg = <0x57>;
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 9d81391..b096988 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -294,6 +294,11 @@
 			pinctrl@6f4000 {
 				compatible = "marvell,ap806-pinctrl";
 				reg = <0x6f4000 0x10>;
+
+				ap_i2c0_pins: i2c-pins-0 {
+					marvell,pins = "mpp4", "mpp5";
+					marvell,function = "i2c0";
+				};
 			};
 
 			gpio0: gpio@6F5040 {
-- 
1.7.9.5

